### PR TITLE
adding fetch limit, aggregate fetch and fields to fetch

### PR DESCRIPTION
**Pull request by mdrobniu**
## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Description
This PR is introducing the following fetch features:
- Fetch limit 
- Aggregate fetch - fetch all events from the query into one fetch. 
- Fields to fetch - allows the user to specify which fields to fetch.

## Screenshots
screenshot

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [X] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No
   - [X] No idea

### CODE REVIEW

**Comment by mdrobniu** [comment]
This was tested on a customer instance.

**Comment by guykeller** [comment]
@mdrobniu 
Is there a possibility to see a demo of this change? Or a video showing it working?

**Comment by guykeller** [comment]
@mdrobniu Just making sure you've seen my latest comment
Would love to take this forward when you have the time

**Comment by mdrobniu** [comment]
@guykeller Sorry I've missed this somehow. I'll record a demo. We can also have an online session.

**Comment by mdrobniu** [comment]
[](https://postimg.cc/qNyNV3b9)

[](https://postimg.cc/PNVpmhvF)

[](https://postimg.cc/N508cpmZ)

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml [hunk 1]
```diff
@@ -468,15 +485,32 @@ script:
         if (params.eventsQuery){
             query = params.eventsQuery;
         }
-        var fields = null;
+        if (params.fields){
+            var fields = params.fields
+            var discover_fields = true;
+        } else {
+            var fields = null;
+            var discover_fields = false
```
Comment (guykeller) [inline Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml:493]: Let's add a semicolon at the end of each line

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml [hunk 2]
```diff
@@ -31,6 +31,23 @@ configuration:
   defaultvalue: ""
   type: 0
   required: false
+- display: Fields
```
Comment (guykeller) [inline Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml:34]: let's add a description for both parameters

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml (hunk 2 above)
Comment (mdrobniu) [reply Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml:34]: don

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml (hunk 1 above)
Comment (mdrobniu) [reply Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml]: done

**Comment by mdrobniu** [comment]
@guykeller Done as requested

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml [hunk 3]
```diff
@@ -468,15 +492,32 @@ script:
...
-            endTime, null, null, null, false, null,
-            null, null, 100, fields);
+            endTime, discover_fields, null, null, false, null,
+            null, null, fetchlimit, fields);
 
         var incidents = [];
-        for (var i = 0; i < events.length; i++) {
-            var incident = incidentFromEvent(events[i]);
+        if(params.aggregate){
+            var events_aggregate = { "Events" : events }
```
Comment (guykeller) [inline Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml:513]:
```suggestion
            var events_aggregate = { "Events" : events };
```

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml [hunk 4]
```diff
@@ -468,15 +492,32 @@ script:
         if (params.eventsQuery){
             query = params.eventsQuery;
         }
-        var fields = null;
+        if (params.fields){
+            var fields = params.fields
```
Comment (guykeller) [inline Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml:496]:
```suggestion
            var fields = params.fields;
```

**Review by guykeller — CHANGES_REQUESTED** [review]

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml (hunk 3 above)
Comment (mdrobniu) [reply Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml:513]: done

File: Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml (hunk 4 above)
Comment (mdrobniu) [reply Packs/ArcSightLogger/Integrations/integration-ArcSightLogger.yml]: done

**Review by guykeller — APPROVED** [review]

**Pull request merged**